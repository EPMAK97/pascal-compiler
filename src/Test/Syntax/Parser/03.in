var a, b : integer;
    c : array [1..10] of integer;
    d : array [1..10, 2..20] of double;
type
    e = integer;
    f = e;
    rec = record
       inRecord : integer;
    end;
const g = 2;
      n = 2 + 2 / 5;
Begin
end.