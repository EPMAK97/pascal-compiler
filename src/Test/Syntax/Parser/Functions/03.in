program test;

procedure p(a: Integer = 10 + 10);
begin
end;

function f(a: Double = 10 * 10.1; const c: Char = 'c') : Double;
begin
	Result := 1.1;
end;

begin
end.