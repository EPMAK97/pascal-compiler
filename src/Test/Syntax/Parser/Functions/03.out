FUNCTION : test
    result : NIL
    function vars {
    FUNCTION : p
        result : NIL
        function params {
        a : INT

        └── 20
        }
        function vars {
        }
        └── statements

    FUNCTION : f
        result : DOUBLE
        function params {
        a : DOUBLE

        └── 101.0
        c : const CHAR

        └── c
        }
        function vars {
        }
        └── statements
                     └── :=
                          ├── Result
                          └── 1.1

    }
    └── statements

